src/MCPClient/lib/clientScripts/archivematicaAssignFileUUID.py

```diff
@@ -17,62 +18,106 @@
 # You should have received a copy of the GNU General Public License
 # along with Archivematica.  If not, see <http://www.gnu.org/licenses/>.
 
-# @package Archivematica
-# @subpackage archivematicaClientScript
-# @author Joseph Perry <joseph@artefactual.com>
+import argparse
+import os
+import re
 import sys
 import uuid
-from optparse import OptionParser
 
 import django
 django.setup()
 # dashboard
-from main.models import File
+from main.models import File, Transfer
 # archivematicaCommon
 from custom_handlers import get_script_logger
 from fileOperations import addFileToTransfer
 from fileOperations import addFileToSIP
 
+import metsrw
+
+
+logger = get_script_logger('archivematica.mcp.client.assignFileUUID')
+
+
+def find_mets_file(unit_path):
+    """
+    Return the location of the original METS in a Archivematica AIP transfer.
+    """
+    p = re.compile(r'^METS\..*\.xml$', re.IGNORECASE)
+    src = os.path.join(unit_path, 'metadata')
+    for item in os.listdir(src):
+        m = p.match(item)
+        if m:
+            return os.path.join(src, m.group())
+
+
+def get_file_uuid_from_mets(sip_directory, file_path_relative_to_sip):
+    """
+    Look up the UUID of the file in the METS document using metsrw.
+    """
+    mets_file = find_mets_file(sip_directory)
+    if not mets_file:
+        logger.info('Archivematica AIP: METS file not found.')
+        return
+    logger.info('Archivematica AIP: reading METS file %s.', mets_file)
+    mets = metsrw.METSDocument.fromfile(mets_file)
+
+    file_path_relative_to_sip = file_path_relative_to_sip.replace('%transferDirectory%', '', 1)
+    file_path_relative_to_sip = file_path_relative_to_sip.replace('%SIPDirectory%', '', 1)
+
+    # Warning! This is not the fastest way to achieve this. But we will focus
+    # on optimizations later.
+    # TODO: is it ok to assume that the file structure is flat?
+    for entry in mets.all_files():
+        if entry.path == file_path_relative_to_sip:
+            logger.info('Archivematica AIP: file UUID of has been found in the METS document (%s).', entry.path)
+            return entry.file_uuid
+    logger.info('Archivematica AIP: file UUID has not been found in the METS document: %s', file_path_relative_to_sip)
+
+
+def main(file_uuid=None, file_path='', date='', event_uuid=None, sip_directory='', sip_uuid=None, transfer_uuid=None, use='original', update_use=True):
+    if file_uuid:
```

Comment:
This and below could also be fixed by starting with `if file_uuid == "None": file_uuid = None`